grails-app/services/org/pih/warehouse/core/IdentifierService.groovy

```diff
@@ -340,9 +338,14 @@ class IdentifierService {
             }
         }
 
-        // If product type is not deafult and has sequential identifier format or code,
-        // then generate sequentional, otherwise random
-        if (productType?.productIdentifierFormat?.contains("0") || productType?.code) {
+        // If product type is not default
+        // and has sequential identifier format containing "0" character which indicates the sequential part
+        // or identifier format is not specified when generatorTypeCode is SEQUENCE but has a code
+        // then generate sequential, otherwise random
+        Boolean shouldGenerateSequential = !productType?.productIdentifierFormat &&
```

Comment:
@alannadolny `!productType?.productIdentifierFormat &&` this part is not required here, since this does not indicate if it needs to be sequential (plus it don't have to be checked at all at this level)